pkg/jobs/registry.go

```diff
@@ -1812,6 +1815,23 @@ func (r *Registry) CheckPausepoint(name string) error {
 	return nil
 }
 
+func (r *Registry) RelocateLease(
+	ctx context.Context,
+	txn isql.Txn,
+	id jobspb.JobID,
+	destID base.SQLInstanceID,
+	destSession sqlliveness.SessionID,
+) (sentinel error, failure error) {
+	if _, err := r.db.Executor().Exec(ctx, "job-relocate-coordinator", txn.KV(),
+		"UPDATE system.jobs SET claim_instance_id = $2, claim_session_id = $3 WHERE id = $1",
+		id, destID, destSession.UnsafeBytes(),
```

Comment:
This feels like another salvo in the battle over whether these job APIs should require that the updater hold the lease currently or not.  In this case, I think I agree with your decision here.

Reply:
Yeah, I don't see why it needs the lease since we're moving the lease itself and not doing anything else.